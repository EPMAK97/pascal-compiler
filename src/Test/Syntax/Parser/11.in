procedure m(var x : array of integer);
const cc = 1;
Begin
end;
procedure sum(a : integer);
var mok : array [1..10] of integer;
type
    cp = ls;
begin
    y := y - 2;
    while x >= 10 do begin
        x := x + 1;
    end;
end;
var m : double;
Begin
    x := x + 1;
    z := z - z;
    for i := 1 to 10 do
      x := x - 1;
    sum := 2 + 2;
end.