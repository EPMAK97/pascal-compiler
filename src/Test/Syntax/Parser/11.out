└── Main Program
    ├── procedure
    │   ├── array
    │   │   └── integer
    │   │       └── x
    │   └── const
    │       └── cc
    │           └── 1
    ├── procedure
    │   ├── integer
    │   │   └── a
    │   ├── var
    │   │   └── array
    │   │       └── mok
    │   │           └── integer
    │   │               ├── 1
    │   │               └── 10
    │   ├── type
    │   │   └── ls
    │   │       └── cp
    │   ├── y
    │   │   └── -
    │   │       ├── y
    │   │       └── 2
    │   └── while
    │       ├── >=
    │       │   ├── x
    │       │   └── 10
    │       └── x
    │           └── +
    │               ├── x
    │               └── 1
    ├── var
    │   └── double
    │       └── m
    ├── x
    │   └── +
    │       ├── x
    │       └── 1
    ├── z
    │   └── -
    │       ├── z
    │       └── z
    ├── for
    │   ├── i
    │   │   ├── 1
    │   │   └── 10
    │   └── x
    │       └── -
    │           ├── x
    │           └── 1
    └── sum
        └── +
            ├── 2
            └── 2
